src/V3Number.cpp

```diff
@@ -883,7 +883,7 @@ string V3Number::toString() const {
     return str;
 }
 
-uint32_t V3Number::toHash() const { return m_value[0]; }
+V3Hash V3Number::toHash() const { return V3Hash(m_width * (m_value[0] | 1)); }
```

Comment:
Can you please run a regression before and after this patch file, and do a recursive compare of the generated C++ code between the versions to make sure nothing is different?  I suspect this change is correct, but want to make sure no optimizations are lost by accident, thanks.

Reply:
Here are the results:

In the regression, all are the same, except:
- Tests having to do with using a prot lib, which are t_prot_lib*, and t_hier_block* (used protlib underneath) have different hash values in the emitted code, but are otherwise the same
- Tests using --protect-ids have different hashes all over. Using eyeballs, they structurally are the same.
- The only meaningful difference is in ./test_regress/t/t_interface_gen5_noinl.pl and 
./test_regress/t/t_interface_gen7_noinl.pl. With the new hashing, these have 1 and 3 fewer output modules respectively. This is due to V3Param now recognizing that interface references are the same (as they now hash the same), which means it does not create multiple  identical parametrizations, so that sound like an improvement. As an example, these 2 instances now create only a single parametrization, while they used to make to before this patch:
https://github.com/verilator/verilator/blob/fd354922262d7b988035557c204789a9908bca49/test_regress/t/t_interface_gen5.v#L23-L39

In the ext tests, all are the same, except for one name change in t_cores_eh2_cmark, where the output is as if 
 you run `sed s/Vtb_top_eh2_exu_alu_ctl__Pz6/Vtb_top_eh2_exu_alu_ctl__Pz1/g`, i.e.: that one identifier changed, but structurally it's all the same.

Reply:
Perfect, thanks for checking.